src/methods/oceanconversions.jl

```diff
@@ -0,0 +1,227 @@
+depends on depth and _latitude_ so the returned pressure is stored as a variable in the
+resulting `Raster` rather than a vertical dimension.
+"""
+function convert_z_to_p(raster::Raster, rs_dims::Tuple)
+
+    lons, lats, z, time = rs_dims
+    p = similar(Array(raster))
+    if isnothing(time)
+        for (i, lon) ∈ enumerate(lons), (j, lat) ∈ enumerate(lats)
+            p[i, j, :] .= GibbsSeaWater.gsw_p_from_z.(z, lat)
+        end
+        rs_dims = (lons, lats, z)
+    else
+        for t ∈ time
+            for (i, lon) ∈ enumerate(lons), (j, lat) ∈ enumerate(lats)
+                p[i, j, :, t] .= GibbsSeaWater.gsw_p_from_z.(z, lat)
+            end
+        end
+    end
+
+    return Raster(p, rs_dims)
+
+end
+
+"""
+    function convert_Sₚ_to_Sₐ(raster::Raster, p::raster)
+Convert a `Raster` of practical salinity (`Sₚ`) to absolute salinity (`Sₐ`) using
+`gsw_sa_from_sp`.
+"""
+function convert_Sₚ_to_Sₐ(Sₚ::Raster, p::Raster, rs_dims::Tuple)
+
+    lons, lats, z, time = rs_dims
+    Sₐ = similar(Array(Sₚ), Union{Float64, Missing})
+    if isnothing(time)
+        for (i, lon) ∈ enumerate(lons), (j, lat) ∈ enumerate(lats)
+
+            Sₚ_profile = Sₚ[X(At(lon)), Y(At(lat))]
+            find_nm = findall(.!ismissing.(Sₚ_profile))
+            p_profile = p[X(At(lon)), Y(At(lat))]
+            Sₐ[i, j, find_nm] = GibbsSeaWater.gsw_sa_from_sp.(Sₚ_profile, p_profile, lon, lat)
+
+        end
+        rs_dims = (lons, lats, z)
+    else
+        for t ∈ time
+            for (i, lon) ∈ enumerate(lons), (j, lat) ∈ enumerate(lats)
+
+                Sₚ_profile = Sₚ[X(At(lon)), Y(At(lat)), Ti(t)]
+                find_nm = findall(.!ismissing.(Sₚ_profile))
+                p_profile = p[X(At(lon)), Y(At(lat)), Ti(t)]
+                Sₐ[i, j, find_nm, t] = GibbsSeaWater.gsw_sa_from_sp.(Sₚ_profile, p_profile, lon, lat)
+
+            end
+        end
+    end
+
+    return Raster(Sₐ, rs_dims)
+
+end
+
+"""
+    function convert_θ_to_Θ(raster::Raster, Sₐ::raster)
+Convert a `Raster` of potential temperature (`θ`) to conservative temperature (`Θ`) using
+`gsw_ct_from_pt`. This conversion depends on absolute salinity.
+"""
+function convert_θ_to_Θ(θ::Raster, Sₐ::Raster, rs_dims::Tuple)
+
+    lons, lats, z, time = rs_dims
+    Θ = similar(Array(θ), Union{Float64, Missing})
+
+    if isnothing(time)
+        for (i, lon) ∈ enumerate(lons), (j, lat) ∈ enumerate(lats)
+
+            Sₐ_profile = Sₐ[X(At(lon)), Y(At(lat))]
+            find_nm = findall(.!ismissing.(Sₐ_profile))
+            θ_profile = θ[X(At(lon)), Y(At(lat)), Z(find_nm)]
+            Θ[i, j, find_nm] = GibbsSeaWater.gsw_ct_from_pt.(Sₐ_profile[find_nm], θ_profile)
+
+        end
+        rs_dims = (lons, lats, z)
+    else
+        for t ∈ time
+            for (i, lon) ∈ enumerate(lons), (j, lat) ∈ enumerate(lats)
+
+                Sₐ_profile = Sₐ[X(At(lon)), Y(At(lat)), Ti(t)]
+                find_nm = findall(.!ismissing.(Sₐ_profile))
```

Comment:
you can pass an anonymous function to `findall` rather than broadcasting it and allocating intermediate arrays

But I think you can just use the BitArray directly and not use `findall`